jobqueue_features/clusters_controller.py

```diff
@@ -63,19 +75,29 @@ def _close(self):
         self._close_clients()
         self._close_clusters()
 
+    def _close_cluster(self, id_):
+        # type: (str) -> None
+        cluster = self._clusters.pop(id_, None)
+        if cluster and type(cluster) is not LocalCluster:
+            cluster.close()
+
     def _close_clusters(self):
         # type: () -> None
-        for cluster in self._clusters.values():
-            if cluster and type(cluster) is not LocalCluster:
-                cluster.close()
-        self._clusters = {"default": None}
+        for id_ in self._clusters.keys():
```

Comment:
Looking at the failing tests, you need to work off a copy of `self._clusters.keys()` since the dictionary changes every time you call `self._close_cluster()`

Reply:
Agree, `self._clusters.keys()` should make it right since it forces to evaluate keys before any next actions.